logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCase.kt

```diff
@@ -0,0 +1,36 @@
+package com.wire.kalium.logic.feature.connection
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.connection.ConnectionRepository
+import com.wire.kalium.logic.data.user.ConnectionState
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.functional.onFailure
+import com.wire.kalium.logic.kaliumLogger
+
+/**
+ * Use Case that allows a user to block user
+ */
+fun interface UnblockUserUseCase {
+    /**
+     * Use case [UnblockUserUseCase] operation
+     *
+     * @param userId the target user whom to unblock
+     * @return a [Either] indicating the operation result:
+     * Left - [CoreFailure] if occurred,
+     * Right - [Unit] is success
+     */
+    suspend operator fun invoke(userId: UserId): Either<CoreFailure, Unit>
```

Comment:
I believe that this return type was supposed to not leak to clients, if I recall correctly from Prev. Android collectives 🤔  We are wrapping the result in a new type.

Reply:
Yepp, it was already agreed that Kalium Use Cases should not expose the `Either` construction to consuming app. The correct way would be to provide a wrapping `Result` class in the same fashion than other use cases do.